public/app/core/internationalization/constants.ts

```diff
@@ -10,61 +11,96 @@ export const PSEUDO_LOCALE = 'pseudo-LOCALE';
 
 export const DEFAULT_LANGUAGE = ENGLISH_US;
 
-interface LanguageDefinitions {
+export type LocaleFileLoader = () => Promise<ResourceKey>;
+
+export interface LanguageDefinition<Namespace extends string = string> {
   /** IETF language tag for the language e.g. en-US */
   code: string;
 
   /** Language name to show in the UI. Should be formatted local to that language e.g. Français for French */
   name: string;
 
   /** Function to load translations */
-  loader: () => Promise<ResourceKey>;
+  loader: Record<Namespace, LocaleFileLoader>;
 }
 
-export const LANGUAGES: LanguageDefinitions[] = [
+export const LANGUAGES: LanguageDefinition[] = [
   {
     code: ENGLISH_US,
     name: 'English',
-    loader: () => import('../../../locales/en-US/grafana.json'),
+    loader: {
+      grafana: () => import('../../../locales/en-US/grafana.json'),
+    },
```

Comment:
With this approach, `grafana-enterprise` is one namespace, but I guess some of this could be extended to plugins?

Reply:
Indeed (see the epilogue in my PR description). It would probably be a bit different to this, but I can start to see how we would use namespaces to segregate and load translations from plugins.